lib/galaxy/webapps/galaxy/controllers/workflow.py

```diff
@@ -948,7 +954,8 @@ def build_from_current_history( self, trans, job_ids=None, dataset_ids=None, dat
                                            url_for( controller='workflow', action='run', id=workflow_id ) ) )
 
     @web.expose
-    def run( self, trans, id, history_id=None, **kwargs ):
```

Comment:
I tried using the 'workflow_dict' method in API to pull the workflow's JSON but it looks a bit different. And using it makes the workflow steps undefined in the run mode. I think we need to merge the calls. I will look into it more.

Workflow dict using controller call ('run_workflow') (which works perfectly in run mode):

![controller_wf](https://user-images.githubusercontent.com/3022518/26984523-f7fbcdc8-4d3f-11e7-9463-655ff74679e6.png)



Workflow dict using api call ('workflow_dict'):

![api_wf](https://user-images.githubusercontent.com/3022518/26984564-1efb6690-4d40-11e7-866a-5791f51bd013.png)